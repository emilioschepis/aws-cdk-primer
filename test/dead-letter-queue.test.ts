import { SynthUtils } from "@aws-cdk/assert";
import { Stack } from "@aws-cdk/core";
import "@aws-cdk/assert/jest";

import dlq = require("../lib/dead-letter-queue");

test("snapshot: dlq creates an alarm", () => {
  const stack = new Stack();
  new dlq.DeadLetterQueue(stack, "DLQ");
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test("unit: dlq creates an alarm", () => {
  const stack = new Stack();
  new dlq.DeadLetterQueue(stack, "DLQ");
  expect(stack).toHaveResource("AWS::CloudWatch::Alarm", {
    MetricName: "ApproximateNumberOfMessagesVisible",
    Namespace: "AWS/SQS",
  });
});
