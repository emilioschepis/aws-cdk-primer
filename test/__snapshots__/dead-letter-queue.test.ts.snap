// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: dlq creates an alarm 1`] = `
Object {
  "Resources": Object {
    "DLQ581697C4": Object {
      "DeletionPolicy": "Delete",
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "DLQAlarm008FBE3A": Object {
      "Properties": Object {
        "AlarmDescription": "There are messages in the Dead Letter Queue",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "QueueName",
            "Value": Object {
              "Fn::GetAtt": Array [
                "DLQ581697C4",
                "QueueName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "ApproximateNumberOfMessagesVisible",
        "Namespace": "AWS/SQS",
        "Period": 60,
        "Statistic": "Maximum",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
  },
}
`;
